File: Day-2/Q6.cpp
#include <iostream>
#include <unordered_set>
#include <vector>
using namespace std;

int count_duplicate_pairs(vector<int> &arr)
{
    int count = 0;
    unordered_set<int> unique_elements(arr.begin(), arr.end());

    for (int num : arr)
    {
        if (unique_elements.find(num + 1) != unique_elements.end())
            count++;
    }

    return count;
}

int main()
{
    int n;
    cin >> n;

    vector<int> arr(n);
    for (int i = 0; i < n; i++)
        cin >> arr[i];

    int result = count_duplicate_pairs(arr);
    cout << result << endl;

    return 0;
}

File: Week-1/Day-2/Q6.py
def count_duplicate_pairs(arr):
    count = 0
    unique_elements = set(arr)

    for num in arr:
        if num + 1 in unique_elements:
            count += 1

    return count

n = int(input())
arr = list(map(int, input().split()))

result = count_duplicate_pairs(arr)
print(result)
